workers/unity/Assets/Fps/Scripts/SetupLogic/AdvancedEntityPipeline.cs

```diff
@@ -6,94 +6,104 @@
                     {
-                        authority = true;
+                        if (attributeSet.Attribute.Contains(workerIdAttribute))
+                        {
+                            authority = true;
+                        }
                     }
-                }
 
-                var serverPosition = entity.GetComponent<ServerMovement.Component>();
-                var position = serverPosition.Latest.Position.ToVector3() + worker.Origin;
+                    var serverPosition = entity.GetComponent<ServerMovement.Component>();
+                    var position = serverPosition.Latest.Position.ToVector3() + worker.Origin;
 
-                var prefab = authority ? cachedAuthPlayer : cachedNonAuthPlayer;
-                var gameObject = Object.Instantiate(prefab, position, Quaternion.identity);
+                    var prefab = authority ? cachedAuthPlayer : cachedNonAuthPlayer;
+                    var gameObject = Object.Instantiate(prefab, position, Quaternion.identity);
 
-                gameObjectsCreated.Add(entity.SpatialOSEntityId, gameObject);
-                gameObject.name = GetGameObjectName(prefab, entity, worker);
-                linker.LinkGameObjectToSpatialOSEntity(entity.SpatialOSEntityId, gameObject, componentsToAdd);
-                return;
+                    gameObjectsCreated.Add(entity.SpatialOSEntityId, gameObject);
+                    gameObject.name = GetGameObjectName(prefab, entity, worker);
+                    linker.LinkGameObjectToSpatialOSEntity(entity.SpatialOSEntityId, gameObject, componentsToAdd);
+                    return;
+                }
             }
-        }
 
-        fallback.OnEntityCreated(entity, linker);
-    }
-
-    private static string GetGameObjectName(GameObject prefab, SpatialOSEntity entity, Worker worker)
-    {
-        return string.Format(GameobjectNameFormat, prefab.name, entity.SpatialOSEntityId, worker.WorkerType);
-    }
+            fallback.OnEntityCreated(entity, linker);
+        }
 
-    public void OnEntityRemoved(EntityId entityId)
-    {
-        if (!gameObjectsCreated.TryGetValue(entityId, out var go))
+        private static string GetGameObjectName(GameObject prefab, SpatialOSEntity entity, Worker worker)
         {
-            fallback.OnEntityRemoved(entityId);
-            return;
+            return string.Format(GameobjectNameFormat, prefab.name, entity.SpatialOSEntityId, worker.WorkerType);
         }
 
-        gameObjectsCreated.Remove(entityId);
-        Object.Destroy(go);
+        public void OnEntityRemoved(EntityId entityId)
+        {
+            if (!gameObjectsCreated.TryGetValue(entityId, out var go))
+            {
+                fallback.OnEntityRemoved(entityId);
+                return;
+            }
+
+            // Trigger a callback when authoritative player gets removed
+            if (go.GetComponent<FpsDriver>() != null)
```

Comment:
This looks a bit hacky, is there no better way to check if this entity was the player authoritative one?

Reply:
after checking , the entity it was linked to is already destroyed. So we can't use this without refactoring some other stuff. Will leave this for now as it is